gobblin-service/src/main/java/gobblin/service/modules/core/GobblinServiceManager.java

```diff
@@ -250,44 +258,60 @@ private Path getServiceWorkDirPath(FileSystem fs, String serviceName, String ser
    * @param changeContext notification context
    */
   private void handleLeadershipChange(NotificationContext changeContext) {
-    if (this.helixManager.isLeader()) {
-      LOGGER.info("Leader notification for {} HM.isLeader {}", this.helixManager.getInstanceName(),
-          this.helixManager.isLeader());
+    if (this.helixManager.isPresent() && this.helixManager.get().isLeader()) {
```

Comment:
yeah we would not get a notification, but Intellij does it not likes it if I invoke helixManager.get() without checking helixManager.isPresent()